consensus/consensus.go

```diff
@@ -104,15 +108,15 @@ type Engine interface {
 	//
 	// Note: The block header and state database might be updated to reflect any
 	// consensus rules that happen at finalization (e.g. block rewards).
-	Finalize(config *params.ChainConfig, header *types.Header, state *state.IntraBlockState, txs []types.Transaction, uncles []*types.Header, r types.Receipts, e EpochReader, chain ChainHeaderReader, syscall SystemCall) error
+	Finalize(config *params.ChainConfig, header *types.Header, state *state.IntraBlockState, txs []types.Transaction, uncles []*types.Header, r types.Receipts, systemTxs *[]*types.Transaction, usedGas *uint64, e EpochReader, chain ChainHeaderReader, syscall SystemCall) error
```

Comment:
Since `types.Transaction` is an interface in Erigon, unlike concrete type in go-ethereum, it should not be used
as `*[]*types.Transaction`, but instead `*[]types.Transaction`. Please change it accordingly in other places. Otherwise it is pointer to the interface, which is unnecessary overhead and less readable.

Also, instead of receiving pointer to the slice `*[]`, so that `Finalize` can modify it, it can return the slice as one of the return values. The only scenarios where I'd think passing pointer to a slice would help is if the slice is expensive to create and needs to be re-used, but I don't think it is the case here.

To summarise, the signature should be:
```
Finalize(config *params.ChainConfig, header *types.Header, state *state.IntraBlockState, txs []types.Transaction, uncles []*types.Header, r types.Receipts, e EpochReader, chain ChainHeaderReader, syscall SystemCall) (systemTxs []types.Transaction, usedGas uint64, err error)
```

then it is clear that `Finalize` is suppose to produce and return a slice of system transaction and also amount of gas used for those (I assume)